### ISSUE TITLE
gtt sync suddenly not working anymore

### ISSUE THREAD

**Issue by mn-martin** [kriskbx/gitlab-time-tracker#15]
It has always worked perfectly fine but now all of a sudden sync keeps getting stuck a 0%. I'm unable to sync and have no idea how to debug or something.

Are there any other users experiencing the same issue?
Can you help me please? This is a really important big project and now I'm stuck in the middle of it ...

```bash
gtt sync
⚙️  Syncing time records... (0/11) [----------------------------------------] 0% - 0m left
```

**Comment by mn-martin** [comment]
Btw. reporting (`gtt report`) still works thought.

**Comment by mn-martin** [comment]
So I managed to find a solution:

1.  I moved the last 10 time records from ~/.gtt/frames to a temporary folder
2. I synced and it did work with the one time record
3. I moved 3 records a time in batches back to ~/.gtt/frames and synced till all 11 records were processed
4. Now everything's back to normal

Be careful when doing this because gtt will not check for unknown files when iterating through the frames directory. Your operating system may create a hidden .DS_Store file which will cause gtt log to fail with Error: undefined. (Simply deleting that .DS_Store file fixed the issue for me thought.)

It seems like one should sync very frequently to prevent issues like that? Shouldn't work that way thought :-/

**Comment by kriskbx** [comment]
You're totally right: gtt doesn't filter out unknown or unreadable files in the frames directory and fails with an error message. This is not intended and I changed it so that unreadable/unknown frames are ignored now.

I personally use a cron that triggers sync automatically every 5 minutes, so I never ran into syncing a bigger amount of frames. I tried it out and discovered a huge bug. Thanks a lot for pointing me in that direction: I would have never encountered it myself.

I just released version 1.6.3 which fixes the bug and makes the sync process more verbose like the reports: so you can see what gtt is actually doing instead of staring at a process bar that doesn't move. ;) Make sure to [update gtt accordingly](https://github.com/kriskbx/gitlab-time-tracker#updating).
